### PR TITLE
refactor grammar

### PR DESCRIPTION
**Pull request by askeko**
Overview of changes:

- Removed assignments, they do not exist in functional languages.
- Changed bin/bool expressions to tags (required for translator impl, should possibly be done for statements too.
- Refactored some rules to make it more simple.

Issues:
Game theory rules are still a bit confused in terms of notation used. We should probably make it more intuitive/simple.
Maybe beneficial to include conditions in actions/strategy as parameters to make them more atomic. You would declare more, but each action/strategy would be simple and probably better to reuse.

### CODE REVIEW

File: GameTheoryLanguage/parser/Gtl.g4 [hunk 1]
```diff
@@ -28,36 +25,58 @@ statement
...
+    | expr op=EQUALS expr               # BooleanExpr
+    | expr op=GREATER expr              # BooleanExpr
+    | expr op=LESS expr                 # BooleanExpr
+    | expr op=LESSEQ expr               # BooleanExpr
+    | expr op=GREATEREQ expr            # BooleanExpr
+    | expr op=NOTEQ expr                # BooleanExpr
+    | expr op=AND expr                  # BooleanExpr
+    | expr op=OR expr                   # BooleanExpr
+    | expr op=XOR expr                  # BooleanExpr
+    | NOT expr                          # LogicalNotExpr
```
Comment (enevoldsen1oo) [inline GameTheoryLanguage/parser/Gtl.g4:48]: "op=NOT"?

File: GameTheoryLanguage/parser/Gtl.g4 [hunk 2]
```diff
@@ -28,36 +25,58 @@ statement
...
+    ;
+
 return
     : RETURN expr? SEMICOLON
     ;
 
+// Control structures
 if
-    : IF LPAR b_expr RPAR THEN LCURL list_of_statements RCURL (elseif | else)?
+    : IF LPAR expr RPAR THEN LCURL (statement)* RCURL (elseif | else)?
```
Comment (enevoldsen1oo) [inline GameTheoryLanguage/parser/Gtl.g4:67]: Skal en if kunne være empty i statement?

File: GameTheoryLanguage/parser/Gtl.g4 [hunk 3]
```diff
@@ -28,36 +25,58 @@ statement
...
     ;
 
 elseif
-    : ELSE IF LPAR b_expr RPAR THEN LCURL list_of_statements RCURL (elseif | else)?
+    : ELSE IF LPAR expr RPAR THEN LCURL (statement)* RCURL (elseif | else)?
```
Comment (enevoldsen1oo) [inline GameTheoryLanguage/parser/Gtl.g4:71]: Tom else if?

File: GameTheoryLanguage/parser/Gtl.g4 (hunk 2 above)
Comment (enevoldsen1oo) [reply GameTheoryLanguage/parser/Gtl.g4:67]: En function tvinges for eksempel til ikke at være tom

File: GameTheoryLanguage/parser/Gtl.g4 [hunk 4]
```diff
@@ -28,36 +25,58 @@ statement
...
-
-// Declarations
-declaration
-    : type ID ASSIGN expr SEMICOLON
-    ;
-
-// Assignment
-assignment
-    : ID ASSIGN expr SEMICOLON
+    : ELSE LCURL (statement)* RCURL
```
Comment (enevoldsen1oo) [inline GameTheoryLanguage/parser/Gtl.g4:75]: Tom else statement?

**Review by enevoldsen1oo — CHANGES_REQUESTED** [review]

File: GameTheoryLanguage/parser/Gtl.g4 (hunk 1 above)
Comment (askeko) [reply GameTheoryLanguage/parser/Gtl.g4:48]: nein, men ved ikke hvorfor :)))

**Review by enevoldsen1oo — APPROVED** [review]

**Pull request merged**